src/language-graphql/printer-graphql.js

```diff
@@ -35,31 +35,44 @@ function genericPrint(path, options, print) {
     }
     case "OperationDefinition": {
       const hasOperation = options.originalText[locStart(node)] !== "{";
+      const hasQueryComment =
+        options.originalText
+          .slice(locStart(node), locEnd(node))
+          .includes("query { #") ||
+        options.originalText
+          .slice(locStart(node), locEnd(node))
+          .includes("query {#") ||
+        options.originalText
+          .slice(locStart(node), locEnd(node))
+          .includes("query  { #");
       const hasName = Boolean(node.name);
-      return [
-        hasOperation ? node.operation : "",
-        hasOperation && hasName ? [" ", print("name")] : "",
-        hasOperation && !hasName && isNonEmptyArray(node.variableDefinitions)
-          ? " "
-          : "",
-        isNonEmptyArray(node.variableDefinitions)
-          ? group([
-              "(",
-              indent([
+      if (!hasQueryComment) {
+        return [
+          hasOperation ? node.operation : "",
+          hasOperation && hasName ? [" ", print("name")] : "",
+          hasOperation && !hasName && isNonEmptyArray(node.variableDefinitions)
+            ? " "
+            : "",
+          isNonEmptyArray(node.variableDefinitions)
+            ? group([
+                "(",
+                indent([
+                  softline,
+                  join(
+                    [ifBreak("", ", "), softline],
+                    path.map(print, "variableDefinitions")
+                  ),
+                ]),
                 softline,
-                join(
-                  [ifBreak("", ", "), softline],
-                  path.map(print, "variableDefinitions")
-                ),
-              ]),
-              softline,
-              ")",
-            ])
-          : "",
-        printDirectives(path, print, node),
-        node.selectionSet ? (!hasOperation && !hasName ? "" : " ") : "",
-        print("selectionSet"),
-      ];
+                ")",
+              ])
+            : "",
+          printDirectives(path, print, node),
+          node.selectionSet ? (!hasOperation && !hasName ? "" : " ") : "",
+          print("selectionSet"),
+        ];
+      }
+      return options.originalText.slice(locStart(node), locEnd(node));
```

Comment:
Have you try replace https://github.com/prettier/prettier/blob/c50f2bbe3075757840bbd7ed941647a9bde84266/src/language-graphql/printer-graphql.js#L22-L34 with this? I think it will fix all graphql issues.